src/android/com/totalpave/cordova/inset/Inset.java

```diff
@@ -0,0 +1,310 @@
+                return;
+            }
+
+            this.$currentInset = result;
+
+            JSONObject update = new JSONObject();
+            try {
+                update.put("type", "update");
+                update.put("id", this.getID());
+                update.put("data", this.$currentInset);
+            }
+            catch (JSONException ex) {
+                throw new RuntimeException(ex);
+            }
+
+            PluginResult response = new PluginResult(Status.OK, update);
+            response.setKeepCallback(true);
+            $callback.sendPluginResult(response);
+        }
+
+        private int $mapMask(int webviewMask) {
+            int insetTypeMask = 0;
+
+            if ((webviewMask & WebviewMask.CAPTION_BAR) != 0) {
+                insetTypeMask |= WindowInsetsCompat.Type.captionBar();
+            }
+
+            if ((webviewMask & WebviewMask.DISPLAY_CUTOUT) != 0) {
+                insetTypeMask |= WindowInsetsCompat.Type.displayCutout();
+            }
+
+            if ((webviewMask & WebviewMask.IME) != 0) {
+                insetTypeMask |= WindowInsetsCompat.Type.ime();
+            }
+
+            if ((webviewMask & WebviewMask.MANDATORY_SYSTEM_GESTURES) != 0) {
+                insetTypeMask |= WindowInsetsCompat.Type.mandatorySystemGestures();
+            }
+
+            if ((webviewMask & WebviewMask.NAVIGATION_BARS) != 0) {
+                insetTypeMask |= WindowInsetsCompat.Type.navigationBars();
+            }
+
+            if ((webviewMask & WebviewMask.STATUS_BARS) != 0) {
+                insetTypeMask |= WindowInsetsCompat.Type.statusBars();
+            }
+
+            if ((webviewMask & WebviewMask.SYSTEM_BARS) != 0) {
+                insetTypeMask |= WindowInsetsCompat.Type.systemBars();
+            }
+
+            if ((webviewMask & WebviewMask.SYSTEM_GESTURES) != 0) {
+                insetTypeMask |= WindowInsetsCompat.Type.systemGestures();
+            }
+
+            if ((webviewMask & WebviewMask.TAPPABLE_ELEMENT) != 0) {
+                insetTypeMask |= WindowInsetsCompat.Type.tappableElement();
+            }
+
+            return insetTypeMask;
+        }
+    }
+
+    private ArrayList<Listener> $listeners;
+    private HashMap<String, Listener> $listenerMap;
+    private final Object $listenerLock = new Object();
+
+    @Override
+    protected void pluginInitialize() {
+        $listeners = new ArrayList<>();
+        $listenerMap = new HashMap<>();
+        
+        ViewCompat.setOnApplyWindowInsetsListener(
+                this.cordova.getActivity().findViewById(android.R.id.content), (v, insetProvider) -> {
+
+                    synchronized($listenerLock) {
+                        for (Listener listener : $listeners) {
+                            listener.onInsetUpdate(insetProvider);
+                        }
+                    }
+
+                    return WindowInsetsCompat.CONSUMED;
+                }
+        );
+    }
```

Comment:
With the changes to Inset construction, will this ever have listeners during pluginInitialize?

Reply:
No, but that's the global listener function so not sure what you're point is.

Reply:
oh right lol my bad

Reply:
That block runs everytime `this.cordova.getActivity().findViewById(android.R.id.content)` has insets applied